docs/assets/diagrams/transfer/02-create-transaction.puml

```diff
@@ -0,0 +1,126 @@
+@startuml
+
+title Create Transaction
+
+hide footbox
+
+autonumber
+
+box "Mobile Device" #Light
+  actor Alice
+  participant "PISP App" as App
+end box
+box "PISP" #Light
+  database Firebase
+  participant "PISP Server" as Server
+end box
+box "Mojaloop" #Light
+  participant "Switch" as Mojaloop
+end box
+
+Alice -> App: I want to transfer money\nto this party
+
+rnote right of App #Light 
+  ""firebase.firestore()""
+  ""  .collection("transactions")""
+  ""  .doc("456")""
+  ""  .set({""
+  ""    userId: "bob101",""
+  ""    payee: {""
+  ""      partyIdInfo: {""
+  ""        partyIdType: "MSISDN",""
+  ""        partyIdentifier: "+1-222-222-2222"""
+  ""      }""
+  ""    }""
+  ""  })""
+end rnote
+
+App -> Firebase ++: Create a new document for the lookup
+App <-- Firebase: ""200 OK""
+
+rnote right of Firebase #Light
+  ""event(""
+  ""  "document.create",""
+  ""  "transactions/456",""
+  ""  <snapshot object>""
+  "")""
+end rnote
+
+Firebase -> Server ++: Notify event for new document
+Firebase <-- Server: ""200 OK""
+deactivate Firebase 
+
+Server -> Mojaloop ++: ""**GET /parties/MSISDN/+1-222-222-2222**""
+Server <-- Mojaloop: ""202 Accepted""
+deactivate Server
+
+Mojaloop -> Mojaloop: Lookup data
+
+rnote left of Mojaloop #Light
+  ""{ ""
+  ""  "party": {""
+  ""    "partyIdInfo": {""
+  ""      "partyIdType": "MSISDN",""
+  ""      "partyIdentifier": "+1-222-222-2222",""
+  ""      "fspId": "fspa"""
+  ""    },""
+  ""    "name": "Alice Alpaca",""
+  ""    "personalInfo": {""
+  ""      "complexName": {""
+  ""        "firstName": "Alice",""
+  ""        "middleName": "K",""
+  ""        "lastName": "Alpaca"""
+  ""      },""
+  ""      "dateOfBirth": "1971-12-25"""
+  ""    }""
+  ""  },""
+  "" "accounts": [""
+  ""    {""
+  ""      "id": "alice.fspa",""
+  ""      "currency": "USD"""
+  ""    }""
+  ""  ]""
+  ""} ""
+end rnote
+
+Mojaloop -> Server ++: ""**PUT /parties/MSISDN/+1-222-222-2222**""
+Mojaloop <-- Server: ""200 OK""
+
+deactivate Mojaloop
+
+rnote left of Server #Light
+  ""firebase.firestore()""
+  ""  .collection("transactions")""
+  ""  .document("456")""
+  ""  set({""
+  ""    payee: {""
+  ""      name: "Alice Alpaca",""
+  ""      partyIdInfo: {""
+  ""        partyIdType: "MSISDN",""
+  ""        partyIdentifier: "+1-222-222-2222",""
+  ""        fspId: "fspa"""
+  ""      }""
+  ""    },""
+  ""    status: "PENDING_PAYEE_CONFIRMATION",""
+  ""  }, { merge: true })""
+end rnote
+
+Server -> Firebase ++: Update transaction status
```

Comment:
Btw, I just realized one remaining issue here...

Upon receiving `PUT /parties/MSISDN/+1-222-222-2222`, the server does not know about the `transactionRequestId` associated with that request, which in this case is equal to `456`.

One way that I can think to solve this problem is by storing a temporary document in Firebase to track the list of `transactionRequestId` that are waiting to receive callback with that party identifier.

Alternatively, we can also imitate how `sdk-scheme-adapter` handles this issue, which is by using redis pub/sub (assuming we want to launch multiple instances of this server; otherwise, we can just keep everything in memory). However, it will require the server to also keep the subscription channels open while waiting for reply.

Any thoughts on this? @jgeewax @looilee @lewisdaly

Reply:
To resolve the current issue, I added a temporary field `partyQuery` which will be removed upon receiving a party lookup result... Not sure whether there's a better way to do that...

Reply:
How about passing "456" as a query param of like:

GET /parties/MSISDN/+1-222-222-2222?transactionRequestId=456

Reply:
I think the problem is about how to identify the `transactionRequestId` upon receiving `PUT /parties/{Type}/{ID}`. Unless we store the mapping somewhere, I don't think passing a `transactionRequestId` in the query param will have any effect (?)

Reply:
Sorry I thought I commented but I think I forgot to click on submit. You can have the Mojaloop switch return the same `transactionRequestId` it received in the PUT request, would that not work? This requires some small changes on the Mojaloop switch side.

Reply:
> Couldn't there be multiple transactions with the same partyQuery ?

Sorry, I think i missed replying to this... Yeah, because at the moment we couldn't differentiate between two different `PUT /parties/...` request



> You can have the Mojaloop switch return the same `transactionRequestId` it received by putting it in the PUT request, would that not work? This requires some small changes on the Mojaloop switch side.

I'm actually not sure about it... would it be possible to do something like this @lewisdaly ?

Reply:
Hmm, that's not going to be possible, but perhaps the issue here is that you are trying to tie together a party lookup and a transaction request before the transaction request even exists.

2 other ideas:
1. You add indexes to the `transactions` collection on `partyIdType` and `partyIdentifier`, and then perform a filter to lookup the `transactions.id` later on
2. You could make a temporary collection, something like `partyLookups` that stores the `transaction.id` and callback state (like you said above)

Reply:
Oh okay, I think currently what I'm doing with `partyQuery` temporary field is basically the same with your first suggestion.